### PR TITLE
Update Agent To Differentiate Key Types

### PR DESCRIPTION
**Pull request by NoMoreFood**
- Adjusted registry entires used by the SSH agent to include a SSH key type in the registry key name.

### CODE REVIEW

**Comment by NoMoreFood** [comment]
I'm investigating the test failure.

**Comment by manojampalam** [comment]
Adding a reference to related conversation - https://github.com/NoMoreFood/openssh-portable/commit/d7b6489bd37d63ce50abf9742bc8edcadc4a1f06

File: contrib/win32/win32compat/ssh-agent/keyagent-request.c [hunk 1]
```diff
@@ -154,7 +174,7 @@ process_add_identity(struct sshbuf* request, struct sshbuf* response, struct age
 	if ((!ConvertStringSecurityDescriptorToSecurityDescriptorW(REG_KEY_SDDL, SDDL_REVISION_1, &sa.lpSecurityDescriptor, &sa.nLength)) ||
 	    sshkey_to_blob(key, &pubkey_blob, &pubkey_blob_len) != 0 ||
 	    convert_blob(con, blob, blob_len, &eblob, &eblob_len, 1) != 0 ||
-	    ((thumbprint = sshkey_fingerprint(key, SSH_FP_HASH_DEFAULT, SSH_FP_DEFAULT)) == NULL) ||
+	    ((thumbprint = get_key_subkey_name(key)) == NULL) ||
```
Comment (bagajjal) [inline contrib/win32/win32compat/ssh-agent/keyagent-request.c:177]: What is the advantage of appending ":key_type_string"?
Don't we need the same logic when we read from the registry to remove the ":key_typ_string"?

File: contrib/win32/win32compat/ssh-agent/keyagent-request.c [hunk 2]
```diff
@@ -179,6 +199,12 @@ process_add_identity(struct sshbuf* request, struct sshbuf* response, struct age
 	if ((success == 0) && reg && thumbprint)
 		RegDeleteKeyExA(reg, thumbprint, KEY_WOW64_64KEY, 0);
 
+	/* delete previous version of thumbprint keys without type */
```
Comment (bagajjal) [inline contrib/win32/win32compat/ssh-agent/keyagent-request.c:202]: Isn't the logic should be before we create the thumbprint?

line 180 on the right hand side adds the thumbprint to registry.
Line 203 deletes the thumbprint added in line 180?

**Comment by bagajjal** [comment]
@NoMoreFood - Could you please add the use case why this change is made?
I tried to look at the reference ( NoMoreFood@d7b6489) but it didn't say why the change is initiated. 
The proposal from @manojampalam discussed is different than what we see here?

**Comment by NoMoreFood** [comment]
@bagajjal I'm going to close this for now.  It was never updated in response to @manojampalam's suggestion.

**Pull request closed without merging**